### PR TITLE
[T3A3][T15-A4]Liu_Peng

### PR DESCRIPTION
**Pull request by lproperty**
Ready for review.

### CODE REVIEW

File: src/seedu/addressbook/commands/AddCommand.java [hunk 1]
```diff
@@ -17,9 +17,9 @@
 
     public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a person to the address book. "
             + "Contact details can be marked private by prepending 'p' to the prefix.\n"
-            + "Parameters: NAME [p]p/PHONE [p]e/EMAIL [p]a/ADDRESS  [t/TAG]...\n"
+            + "Parameters: NAME [p]p/PHONE [p]e/EMAIL [p]a/BLOCK, STREET, UNIT, POSTAL_CODE [t/TAG]...\n"
```
Comment (sooyj) [inline src/seedu/addressbook/commands/AddCommand.java:20]: Good update on the help messages!

File: src/seedu/addressbook/data/person/Address.java [hunk 2]
```diff
@@ -1,39 +1,69 @@
...
+    	String trimmedAddress = address.trim();
+    	this.isPrivate = isPrivate;
+
+    	String[] addressElementsString = trimmedAddress.split(ADDRESS_SPLIT_REGEX);
+
+    	if(!isValidAddress(addressElementsString)) {
+    		throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
+
+    	}
+    	this.addressBlock = new AddressBlock(addressElementsString[0]);
```
Comment (sooyj) [inline src/seedu/addressbook/data/person/Address.java:43]: Two suggestions:

1. Use of constants to replace magic numbers.
2. Dont forget SLAP. The trimming, split and new object creation can be placed in a next level method to keep this method readable and clean.

File: src/seedu/addressbook/data/person/Address.java [hunk 3]
```diff
@@ -1,39 +1,69 @@
...
      */
-    public static boolean isValidAddress(String test) {
-        return test.matches(ADDRESS_VALIDATION_REGEX);
+    public static boolean isValidAddress(String[] test) {
+    	if(test.length != 4){
+    		return false;
+    	}
+
+    	for(int i=0;i<4;i++){
+    		if(!test[i].matches(ADDRESS_VALIDATION_REGEX)){
```
Comment (sooyj) [inline src/seedu/addressbook/data/person/Address.java:61]: Good try! Can consider delegating these checks to the respective classes.

File: src/seedu/addressbook/data/person/AddressBlock.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
+package seedu.addressbook.data.person;
+
+/**
```
Comment (sooyj) [inline src/seedu/addressbook/data/person/AddressBlock.java:3]: Good comment for class header.

File: src/seedu/addressbook/data/person/AddressBlock.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+ * Represents a Person's address's block in the address book.
+ */
+
+public class AddressBlock {
+
+	public final String value;
+
+	public AddressBlock (String value) {
```
Comment (sooyj) [inline src/seedu/addressbook/data/person/AddressBlock.java:11]: Consider giving each method a simple comment.

**Review by sooyj — COMMENTED** [review]
An ok change.

Four classes were added, Address class' methods were updated accordingly. However, the four classes are quite limited in functionalities. Can consider adding the validation code in those classes.

Help messages were updated, but not the user guide nor the test cases.

**Comment by lproperty** [comment]
Thank you! :)

**Pull request closed without merging**